src/loaders/jobs/taskfarmer/task_generator.py

```diff
@@ -0,0 +1,247 @@
+    batch_script += '#SBATCH -C cpu\n\n'
+    batch_script += f'cd {job_dir}\n'
+    batch_script += 'export THREADS=32\n\n'
+    batch_script += f'runcommands.sh {task_list_file}'
+
+    batch_script_file = os.path.join(job_dir, 'submit_taskfarmer.sl')
+    with open(batch_script_file, "w") as f:
+        f.write(batch_script)
+
+    return batch_script_file
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='PROTOTYPE - Create the required documents/scripts for the TaskFarmer Workflow Manager.')
+    required = parser.add_argument_group('required named arguments')
+    optional = parser.add_argument_group('optional arguments')
+
+    # Required flag arguments
+    required.add_argument('--tool', required=True, type=str, choices=TOOLS_AVAILABLE,
+                          help='Name of tool to be executed. (e.g. gtdb_tk, checkm2, etc.)')
+
+    required.add_argument(f'--{loader_common_names.KBASE_COLLECTION_ARG_NAME}', required=True, type=str,
+                          help=loader_common_names.KBASE_COLLECTION_DESCR)
+
+    required.add_argument(f'--{loader_common_names.LOAD_VER_ARG_NAME}', required=True, type=str,
+                          help=loader_common_names.LOAD_VER_DESCR)
+
+    required.add_argument('--source_data_dir', required=True, type=str,
+                          help='Source data (genome files) directory. '
+                               '(e.g. /global/cfs/cdirs/kbase/collections/sourcedata/GTDB/r207')
+
+    # Optional arguments
+    optional.add_argument('--root_dir', type=str, default=loader_common_names.ROOT_DIR,
+                          help=f'Root directory for the collections project. (default: {loader_common_names.ROOT_DIR})')
+
+    optional.add_argument('--submit_job', action='store_true', help='Submit job to slurm')
+    optional.add_argument('--no_submit_job', dest='submit_job', action='store_false', help='Do not submit job to slurm')
+
+    args = parser.parse_args()
+
+    tool = args.tool
+    kbase_collection = getattr(args, loader_common_names.KBASE_COLLECTION_ARG_NAME)
+    load_ver = getattr(args, loader_common_names.LOAD_VER_ARG_NAME)
+    source_data_dir = args.source_data_dir
+    root_dir = args.root_dir
+
+    current_datetime = datetime.datetime.now()
+    job_dir = os.path.join(root_dir, 'task_farmer_jobs', f'{tool}_{current_datetime.strftime("%Y_%m_%d_%H_%M_%S")}')
+    os.makedirs(job_dir, exist_ok=True)
+
+    image_str = _fetch_image(REGISTRY, tool)
+    wrapper_file = _create_shifter_wrapper(job_dir, image_str)
+
+    task_list_file, n_jobs = _create_task_list(source_data_dir, kbase_collection, load_ver, tool, wrapper_file, job_dir,
+                                               root_dir)
+
+    batch_script = _create_batch_script(job_dir, task_list_file, n_jobs)
+
+    if args.submit_job:
+        std_out, std_err, returncode = _run_command(['sbatch', os.path.join(job_dir, 'submit_taskfarmer.sl')])
+        print(f'Job submitted to slurm.\n{std_out.strip()}')
```

Comment:
What if returncode  != 0?

Reply:
`_run_command` by default checks if returncode != 0. It will raise an error.

Reply:
:+1: